lib/auth.ts

```diff
@@ -133,6 +152,82 @@ export class Auth {
     return token;
   };
 
+  /**
+   * Perform a request against Kontist REST API
+   */
+  private request = async (path: string, method: HttpMethod, body?: string) => {
+    if (!this.token) {
+      throw new UserUnauthorizedError();
+    }
+
+    const requestUrl = new URL(path, this.baseUrl).href;
+
+    const response = await fetch(requestUrl, {
+      method,
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.token.accessToken}`
+      },
+      body
+    });
+
+    if (!response.ok) {
+      throw new KontistSDKError({
+        status: response.status,
+        message: response.statusText
+      })
+    }
+
+    return response.json();
+  };
+
+  /**
+   * Called periodically by `getMFAConfirmedToken` until
+   * the challenge expires or its status is updated
+   */
+  private pollChallengeStatus = (
+    pendingChallenge: Challenge,
+    resolve: Function,
+    reject: Function
+  ) => async () => {
+    const challenge = await this.request(
+      `${MFA_CHALLENGE_PATH}/${pendingChallenge.id}`,
+      HttpMethod.GET
+    );
+
+    if (new Date(challenge.expiresAt) < new Date()) {
+      clearInterval(this.challengePollIntervalId as IntervalID);
```

Comment:
this line is repeated multiple times, would it be useful to extract it to a separate method?